prober/grpc_health_v1.go

```diff
@@ -0,0 +1,50 @@
+package prober
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/prometheus/blackbox_exporter/config"
+
+	"google.golang.org/grpc"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+)
+
+func ProbeGRPCHEALTHV1(ctx context.Context, target string, module config.Module, registry *prometheus.Registry, logger log.Logger) bool {
+	var (
+		conn    *grpc.ClientConn
+		parts   []string
+		service string
+		host    string
+	)
+
+	parts          = strings.Split(target, "/")
+	host, service  = parts[0], parts[1]
+	conn, err     := grpc.Dial(host, grpc.WithInsecure())
+
+	if err != nil {
+		level.Error(logger).Log("msg", "failed to connect")
+		return false
+	}
+
+	defer conn.Close()
+
+	req := &healthpb.HealthCheckRequest{Service: service}
+	client := healthpb.NewHealthClient(conn)
+	res, err := client.Check(ctx, req)
+
+	if err != nil {
+		level.Error(logger).Log("msg", err)
+		return false
+	}
+
+	if res.Status != healthpb.HealthCheckResponse_SERVING {
```

Comment:
now i've looked deeper, from this line to the rest of the function could be replaced with:
```
return res.Status == healthpb.HealthCheckResponse_SERVING
```

Not sure if that is any clearer though. Up to you.